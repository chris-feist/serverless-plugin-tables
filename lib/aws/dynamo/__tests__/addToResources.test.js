const addToResources = require('../addToResources');

const { createTableResource } = require('../testUtils');

beforeEach(() => {
  jest.clearAllMocks();
});

describe('addToResources', () => {
  test('adds to resources', () => {
    const plugin = {
      service: {
        resources: {},
      },
    };
    const tables = []; // ignored
    const processedTables = [
      createTableResource({ resourceName: 'Res1', tableName: 'Table1' }),
      createTableResource({ resourceName: 'Res2', tableName: 'Table2' }),
      createTableResource({ resourceName: 'Res3', tableName: 'Table3' }),
      createTableResource({ resourceName: 'Res4', tableName: 'Table4' }),
    ];

    addToResources(plugin, tables, processedTables);

    expect(plugin.service.resources.Resources).toBeDefined();
    expect(plugin.service.resources.Resources.Res1).toBeDefined();
    expect(plugin.service.resources).toMatchSnapshot();
  });


  test('initializes resources', () => {
    const plugin = {
      service: {
        resources: {},
      },
    };
    const tables = []; // ignored
    const processedTables = [createTableResource()];

    addToResources(plugin, tables, processedTables);

    expect(plugin.service.resources.Resources).toBeDefined();
    expect(plugin.service.resources).toMatchSnapshot();
  });
});
