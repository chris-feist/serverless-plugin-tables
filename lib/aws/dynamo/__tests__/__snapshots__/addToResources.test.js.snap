// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addToResources adds to resources 1`] = `
Object {
  "Resources": Object {
    "Res1": Object {
      "Properties": Object {
        "TableName": "Table1",
      },
      "Type": "AWS::DynamoDB::Table",
    },
    "Res2": Object {
      "Properties": Object {
        "TableName": "Table2",
      },
      "Type": "AWS::DynamoDB::Table",
    },
    "Res3": Object {
      "Properties": Object {
        "TableName": "Table3",
      },
      "Type": "AWS::DynamoDB::Table",
    },
    "Res4": Object {
      "Properties": Object {
        "TableName": "Table4",
      },
      "Type": "AWS::DynamoDB::Table",
    },
  },
}
`;

exports[`addToResources initializes resources 1`] = `
Object {
  "Resources": Object {
    "TestResource": Object {
      "Properties": Object {
        "TableName": "TestTable",
      },
      "Type": "AWS::DynamoDB::Table",
    },
  },
}
`;
